src/k8s/pkg/utils/k8s/status.go

```diff
@@ -9,24 +9,30 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// WaitApiServerReady waits until the kube-apiserver becomes available.
+func (c *Client) listNodes(ctx context.Context) (*v1.NodeList, error) {
+	nodes, err := c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list nodes: %v", err)
+	}
+	return nodes, nil
+}
+
 func (c *Client) WaitApiServerReady(ctx context.Context) error {
 	return control.WaitUntilReady(ctx, func() (bool, error) {
-		// TODO: use the /readyz endpoint instead
-
-		_, err := c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
-		// We want to retry if an error occurs (=API server not ready)
-		// returning the error would abort, thus checking for nil
+		_, err := c.listNodes(ctx)
 		return err == nil, nil
 	})
 }
 
-// ClusterReady checks the status of all nodes in the Kubernetes cluster.
-// If at least one node is in READY state it will return true.
-func (c *Client) ClusterReady(ctx context.Context) (bool, error) {
-	nodes, err := c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+func (c *Client) IsClusterReady(ctx context.Context) (bool, error) {
+	nodes, err := c.listNodes(ctx)
+
 	if err != nil {
-		return false, fmt.Errorf("failed to list nodes: %v", err)
+		return false, err
+	}
+
+	if len(nodes.Items) == 0 {
+		return false, fmt.Errorf("cluster has no nodes")
 	}
```

Comment:
This is problematic. Why do we consider it an "error" if we got back a response but no kubelets have registered yet? Isn't this handled by the `return false, nil` at the end of the function?

Reply:
Yep, you're right.